alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.alloydb;
+
+import static com.google.common.truth.Truth.assertThat;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.alloydb.v1alpha.InstanceName;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.rpc.Code;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.Socket;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ConnectorTest {
+
+  private static final String INSTANCE_NAME =
+      "projects/<PROJECT>/locations/<REGION>/clusters/<CLUSTER>/instances/<INSTANCE>";
+  private static final String PRIVATE_IP = "127.0.0.2";
+  private static final String SERVER_MESSAGE = "HELLO";
+  private static final String ERROR_MESSAGE_NOT_FOUND = "Resource 'instance' was not found";
+
+  ListeningScheduledExecutorService defaultExecutor;
+
+  @Before
+  public void setUp() throws Exception {
+    defaultExecutor = MoreExecutors.listeningDecorator(Executors.newScheduledThreadPool(8));
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    defaultExecutor.shutdownNow();
+  }
+
+  @Test
+  public void create_successfulPrivateConnection()
+      throws IOException, InterruptedException, NoSuchAlgorithmException, InvalidKeySpecException {
+    FakeSslServer sslServer = new FakeSslServer(SERVER_MESSAGE);
+    sslServer.start(PRIVATE_IP);
+
+    ConnectionConfig config =
+        new ConnectionConfig.Builder().withInstanceName(InstanceName.parse(INSTANCE_NAME)).build();
+
+    Connector connector = newConnector(config.getConnectorConfig());
```

Comment:
nit: passing in the mock as a second argument makes the relationship between the fake SSL server and the connector clean in my opinion.

Reply:
updated.